cli/azd/pkg/alphafeatures/alpha_feature_manager.go

```diff
@@ -0,0 +1,91 @@
+package alphafeatures
+
+import (
+	"log"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+)
+
+type AlphaFeatureManager struct {
+	configManager   config.UserConfigManager
+	userConfigCache config.Config
+	// used for mocking alpha features on testing
+	alphaFeaturesResolver func() []AlphaFeature
+}
+
+func NewAlphaFeaturesManager(configManager config.UserConfigManager) *AlphaFeatureManager {
+	return &AlphaFeatureManager{
+		configManager: configManager,
+	}
+}
+
+func (m *AlphaFeatureManager) ListFeatures() (map[string]AlphaFeature, error) {
+	result := make(map[string]AlphaFeature)
+
+	var alphaFeatures []AlphaFeature
+	if m.alphaFeaturesResolver != nil {
+		alphaFeatures = m.alphaFeaturesResolver()
+	} else {
+		alphaFeatures = mustUnmarshalAlphaFeatures()
+	}
+
+	for _, aFeature := range alphaFeatures {
+		// cast is safe here from string to AlphaFeatureId
+		status := disabledText
+		if m.IsEnabled(AlphaFeatureId(aFeature.Id)) {
+			status = enabledText
+		}
+
+		result[aFeature.Id] = AlphaFeature{
+			Id:          aFeature.Id,
+			Description: aFeature.Description,
+			Status:      status,
+		}
+	}
+
+	return result, nil
+}
+
+func (m *AlphaFeatureManager) IsEnabled(featureId AlphaFeatureId) bool {
+	if m.userConfigCache == nil {
+		config, err := m.configManager.Load()
+		if err != nil {
+			log.Panic("Can't load user config!! %w", err)
+		}
+		m.userConfigCache = config
+	}
+
+	//check if all features is ON
+	if allOn := isEnabled(m.userConfigCache, string(AllId)); allOn {
+		return true
+	}
+
+	// check if the feature is ON
+	if featureOn := isEnabled(m.userConfigCache, string(featureId)); featureOn {
+		return true
+	}
+
+	return false
+}
+
+func isEnabled(config config.Config, id string) bool {
+	longKey := strings.Join([]string{
+		string(parentKey),
+		id,
+	}, ".")
+	value, exists := config.Get(longKey)
+	if !exists {
+		return exists
+	}
+	// safe cast -> reading from config text file
+	stringValue, _ := value.(string)
```

Comment:
We should `panic` with a better error if this conversion fails. The user could have set this to an `int` or a `json` object value

It should read something like:

"Invalid value for 'alpha.feature': %s"